docs/content/chunker.md

```diff
@@ -0,0 +1,237 @@
+last one which may have less data. If file size is unknown in advance,
+chunker will transparently create a temporary copy, record its size
+and repeat the above process. When upload is complete, temporary chunk
+files are given their final names.
+
+By default chunks names are `BIG_FILE_NAME.rclone-chunk.001`,
+`BIG_FILE_NAME.rclone-chunk.002` etc, because the default chunk name
+format is `*.rclone-chunk.###`. You can configure another name format
+using the `--chunker-name-format` option. The format uses asterisk
+`*` as a placeholder for the base file name and one or more consecutive
+hash characters `#` as a placeholder for the chunk number. There must be
+one and only one asterisk. The number of consecutive hashes defines the
+minimum left of a chunk number. If a chunk number has less digits than
+the number of hashes, it is left-padded by zeros. If there are more
+digits in the number, they are left as is. The numbering by default
+start from 1 but there is another option that allows user to start from 0.
+
+For example, if name format is `big_*-##.part`, and original file was
+named `data.txt` and numbering starts from 0, then the first chunk will be
+named `big_data.txt-00.part`, the 99th chunk will be `big_data.txt-98.part`
+and the 302nd chunk will be `big_data.txt-301.part`.
+
+Possible chunk files are ignored if their name does not match given format.
+
+#### Metadata
+
+By default chunker creates metadata object besides data chunks. It is named
+after the original file. Currenly then only supported format is `simplejson`,
+which keeps the following fields:
+
+- Name         - name of the composite file (equal to the meta file name)
+- Size         - total size of all chunks
+- PublicKey    - currently not used, always "null"
+- CreationDate - last modification time
+
+You can disable meta objects by setting meta format option to `none`.
+Without meta object chunker will search current directory for all files
+that follow given chunk name format, collect them into groups by
+detecting chunks with the same base name and show group names as virtual
+composite files. When user requests download, chunker will and transparently
+assemble composite files from chunks. This method is more prone to errors
+due to missing chunks (especially missing last chunk) than the former method.
+
+The list command might encounter files with missinng or invalid chunks.
+This boolean flag tells what rclone should do in such cases. By default
+when chunker detects a missing chunk it will silently ignore the whole
+group. You can use the `--chunker-fail-on-bad-chunks` option to make it
+fail loudly.
+
+#### Modified time and hashes
+
+Chunker stores modification times using the underlying remote so support
+depends on that. Hashes are not stored.
+
+### Limitations ###
+
+Chunker overlay internally streams a file being split or merged and
+consequently fails if an underlying remote switches to multipart mode,
```

Comment:
Why does it fail?  I would have thought it would work properly whether the underlying remote is using chunked upload or not?

It is probably good advice to turn off the chunking though.

Reply:
The user report and logs can be found here: https://github.com/ivandeex/rclone/issues/19
I believe the Box layer was first to manifest the bug because its multipart cutoff is small compared to other backends.

Reply:
Interestingly it manages to upload the entire first chunk OK, but fails on the second chunk with an EOF error. I don't see why that should be unless chunker is returning the EOF on a Read call?

```
019/08/04 07:49:32 DEBUG : mnt/engram/projects/archive/INDIVGraphTheory.tar.gz__tmp.1564919310.001: Multipart upload session started for 64 parts of size 32M
2019/08/04 07:49:32 DEBUG : mnt/engram/projects/archive/INDIVGraphTheory.tar.gz__tmp.1564919310.001: Cancelling multipart upload: multipart upload failed to read source: EOF
2019/08/04 07:49:33 DEBUG : mnt/engram/projects/archive/INDIVGraphTheory.tar.gz: Received error: multipart upload failed to read source: EOF - low level retry 1/10
```

Reply:
fixed in the next round of commits